estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,65 @@
+from odoo import _, api, fields, models
+from datetime import date, timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Offers that propeties receive."
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        copy=False, selection=[("accepted", "Accepted"), ("refused", "Refused")]
+    )
+    partner_id = fields.Many2one("res.partner", required=True, string="Partner")
+    property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+
+    property_type_id = fields.Many2one(
+        related="property_id.property_type_id", store=True
+    )
+    _sql_constraints = [
+        ("positive_offer_price", "CHECK(price > 0)", "Offer Price Must be Positive")
+    ]
+
+    @api.depends("property_id.create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.property_id.create_date + timedelta(
+                    days=record.validity
+                )
+            else:
+                record.date_deadline = date.today() + timedelta(days=record.validity)
```

Comment:
```suggestion
       record.date_deadline = record.property_id.create_date + relativedelta(days=record.validity)
```
can you give it a try?